litert/vendors/qualcomm/core/wrappers/quantize_params_wrapper.cc

```diff
@@ -105,6 +125,30 @@ AxisScaleOffsetQuantizeParamsWrapper::AxisScaleOffsetQuantizeParamsWrapper(
       scale_offsets_.data();
 }
 
+bool AxisScaleOffsetQuantizeParamsWrapper::operator==(
+    const AxisScaleOffsetQuantizeParamsWrapper& other) const {
+  if (qnn_quantize_param_.encodingDefinition !=
+      other.qnn_quantize_param_.encodingDefinition)
+    return false;
+  if (qnn_quantize_param_.quantizationEncoding !=
+      other.qnn_quantize_param_.quantizationEncoding)
+    return false;
+  if (qnn_quantize_param_.axisScaleOffsetEncoding.axis !=
+      other.qnn_quantize_param_.axisScaleOffsetEncoding.axis)
+    return false;
+  if (qnn_quantize_param_.axisScaleOffsetEncoding.numScaleOffsets !=
+      other.qnn_quantize_param_.axisScaleOffsetEncoding.numScaleOffsets)
+    return false;
+
+  return std::equal(
+      qnn_quantize_param_.axisScaleOffsetEncoding.scaleOffset,
+      qnn_quantize_param_.axisScaleOffsetEncoding.scaleOffset +
+          qnn_quantize_param_.axisScaleOffsetEncoding.numScaleOffsets,
+      other.qnn_quantize_param_.axisScaleOffsetEncoding.scaleOffset,
```

Comment:
Also need to specify the range for `other`.
```suggestion
      other.qnn_quantize_param_.axisScaleOffsetEncoding.scaleOffset,
      other.qnn_quantize_param_.axisScaleOffsetEncoding.scaleOffset +
      other.qnn_quantize_param_.axisScaleOffsetEncoding.numScaleOffsets,
```